File: 240517/크기가 n인 순열/n-permutation.py
n = int(input())
visited = [0] * (n+1)

def back(cnt, sm):

    if cnt >= n:
        print(*sm)
        return
    
    for i in range(1, n+1):
        if visited[i] == 0:
            visited[i] = 1
            back(cnt+1, sm+[i])
            visited[i] = 0

back(0, [])

File: 240517/크기가 n인 순열/n-permutation.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim()
const N = parseInt(input)
let visited = Array(N+1).fill(0)

function backtracking(cnt, sm) {

    if (cnt >= N) {
        console.log(...sm)
        return
    }

    for (let i = 1; i <= N; i++) {
        if (visited[i] === 0) {
            visited[i] = 1
            sm.push(i)
            backtracking(cnt+1, sm)
            sm.pop()
            visited[i] = 0
        }

    }
}

backtracking(0, [])